src/main/java/io/sharedstreets/data/SharedStreetsOSMMetadata.java

```diff
@@ -42,6 +46,17 @@ public WaySectionMetadata( WaySection section, boolean storeWaySegmentNames) {
             for(int i = 0; i < section.nodes.length; i++) {
                 this.nodeIds[i] = section.nodes[i].nodeId;
             }
+
+            this.tagNames = new String[section.fields.keySet().size()];
+            this.tagValues = new String[section.fields.keySet().size()];
+
+            int i = 0;
+            for(String key : section.fields.keySet()) {
+                this.tagNames[i] = key;
+                this.tagValues[i] = section.fields.get(key);
+                i++;
+            }
+
```

Comment:
As adding osm tags will have a substantial consequence on size of the generated tiles, could you elaborate a bit on the reasons why these are added? Perhaps this should be optional? Or at least the keys be filterable?